tests/update/tests.py

```diff
@@ -225,6 +225,33 @@ def test_update_with_joined_field_annotation(self):
                             new_name=annotation,
                         ).update(name=F("new_name"))
 
+    def test_update_ordered_by_m2m_aggregation_annotation(self):
+        msg = (
+            "Cannot update when ordering on an aggregation annotation: "
+            "Count(Col(update_bar_m2m_foo, update.Bar_m2m_foo.foo))"
+        )
+        with self.assertRaisesMessage(FieldError, msg):
+            Bar.objects.annotate(m2m_count=Count("m2m_foo")).order_by(
+                "m2m_count"
+            ).update(x=2)
+
+    @unittest.skipIf(
+        connection.vendor == "mysql",
+        (
+            "UPDATE...ORDER BY syntax for MySQL/MariaDB does not support ordering"
+            "related fields"
+        ),
+    )
+    def test_update_ordered_by_m2m_annotation(self):
+        foo = Foo.objects.create(target="bbb")
+        Bar.objects.create(foo=foo)
+
+        Bar.objects.order_by(Abs("m2m_foo")).update(x=2)
+        self.assertEqual(Bar.objects.get().x, 2)
```

Comment:
This works without the patch so we should move it to a separate commit and test method.